explorer/src/components/WorldMap.tsx

```diff
@@ -1,111 +1,80 @@
-import React, { useEffect, useState } from 'react';
-import { geoEqualEarth, geoPath } from 'd3-geo';
-import { feature } from 'topojson-client';
-import { Feature, FeatureCollection, Geometry } from 'geojson';
-// import './styles.css';
+import React from 'react';
+import { scaleLinear } from 'd3-scale';
+import { ComposableMap, Geographies, Geography } from 'react-simple-maps';
+import ReactTooltip from 'react-tooltip';
+import { Box } from '@mui/material';
+import { MainContext } from '../context/main';
+import { countriesData } from '../data/countriesData';
+import { ContentCard } from './ContentCard';
 
-const uuid = require('react-uuid');
-
-const cx = 400;
-const cy = 150;
+const geoUrl =
+  'https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/topojson-maps/world-110m.json';
 
 export const WorldMap: React.FC = () => {
-  const [geographies, setGeographies] = useState<
-    [] | Array<Feature<Geometry | null>>
-  >([]);
-
-  const [scale, setScale] = useState<number>(200);
-  const [windowResizing, setWindowResizing] = useState<boolean>(false);
-
-  const projection = geoEqualEarth()
-    .scale(scale)
-    .translate([cx, cy])
-    .rotate([0, 0]);
-
-  const fetchGeographicData = (): void => {
-    fetch('/data/world-110m.json').then((response) => {
-      if (response.status !== 200) {
-        console.log('Houston, we have a problem');
-        return;
-      }
-      response.json().then((worldData) => {
-        const mapFeatures: Array<Feature<Geometry | null>> = (
-          feature(
-            worldData,
-            worldData.objects.countries,
-          ) as unknown as FeatureCollection
-        ).features;
-        setGeographies(mapFeatures);
-      });
-    });
-  };
-
-  const renderMap = () => {
-    console.log('render the map...');
-  };
-
-  useEffect(() => {
-    fetchGeographicData();
-    renderMap();
-  }, []);
-
-  useEffect(() => {
-    let timeout: any;
-    const handleResize = () => {
-      clearTimeout(timeout);
-
-      setWindowResizing(true);
-
-      timeout = setTimeout(() => {
-        setWindowResizing(false);
-      }, 200);
-    };
-    window.addEventListener('resize', handleResize);
-
-    return () => window.removeEventListener('resize', handleResize);
+  const [tooltipContent, setTooltipContent] = React.useState<string>('');
+  const [data, setData] = React.useState<Record<string, unknown>[]>([]);
```

Comment:
Define an `interface Data {...}` and say what the fields of each item will be will be, then do:

```suggestion
const [data, setData] = React.useState<Data[]>([]);
```